libs/api-catalogue/services/src/lib/restmetadata.service.ts

```diff
@@ -71,26 +78,23 @@ export class RestMetadataService {
           service.description = spec.info.description ?? ''
           service.data = union(service.data, spec.info['x-category'])
           service.pricing = union(service.pricing, spec.info['x-pricing'])
-          service.versions.push({
-            versionId: sorted[i].serviceCode!,
-            details: [
-              {
-                title: spec.info.title,
-                summary: '', // TODO: We should have a short summary
-                description: spec.info.description ?? '',
-                type: TypeCategory.REST,
-                data: spec.info['x-category'] ?? [],
-                pricing: spec.info['x-pricing'] ?? [],
-                links: {
-                  responsibleParty:
-                    spec.info['x-links']?.responsibleParty ?? '',
-                  bugReport: spec.info['x-links']?.bugReport ?? '',
-                  documentation: spec.info['x-links']?.documentation ?? '',
-                  featureRequest: spec.info['x-links']?.featureRequest ?? '',
-                },
-                xroadIdentifier: sorted[i],
-              },
-            ],
+
+          // TODO: This needs to be environment aware
+          service.environments[0].details.push({
+            version: parseVersionNumber(sorted[i].serviceCode!),
+            title: spec.info.title,
+            summary: '', // TODO: We should have a short summary
+            description: spec.info.description ?? '',
+            type: TypeCategory.REST,
+            data: spec.info['x-category'] ?? [],
+            pricing: spec.info['x-pricing'] ?? [],
+            links: {
+              responsibleParty: spec.info['x-links']?.responsibleParty ?? '',
+              bugReport: spec.info['x-links']?.bugReport ?? '',
+              documentation: spec.info['x-links']?.documentation ?? '',
+              featureRequest: spec.info['x-links']?.featureRequest ?? '',
+            },
```

Comment:
I guess this will become environment aware at a later time?

Reply:
Yes, Guðjón is working on that feature in the XRoad Collector.
So I was just making necessary changes so it wouldn't break.